cylc/flow/run_modes/nonlive.py

```diff
@@ -0,0 +1,59 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilities supporting all nonlive modes
+"""
+from typing import TYPE_CHECKING, Dict, List
+
+from cylc.flow import LOG
+from cylc.flow.run_modes.skip import check_task_skip_config
+from cylc.flow.task_state import RunMode
+
+if TYPE_CHECKING:
+    from cylc.flow.taskdefs import TaskDefs
```

Comment:
This doesn't exist. Did you mean
```suggestion
    from cylc.flow.taskdef import TaskDef
```